presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -248,24 +247,24 @@ private void checkAllFuturesDone()
     }
 
     @Override
-    public CompletableFuture<LookupSource> readSpilledLookupSource(Session session, int partition)
+    public synchronized CompletableFuture<LookupSource> readSpilledLookupSource(Session session, int partition)
```

Comment:
The method as it stands now does only two things:

* `spilledLookupSources.get(partition)`
* `lookupSourceSpiller.getAllSpilledPages()`

the rest is a callback.

For me, it's not obvious what synchronization is needed for. Or -- maybe it is needed, but this fact is not related to the rest of the commit?